letterparser/utils.py

```diff
@@ -0,0 +1,23 @@
+# coding=utf-8
+
+"utility helper functions"
+
+
+def unicode_encode(string):
+    """safely encode string as utf8 by catching exceptions"""
+    if string is None or isinstance(string, str):
+        return string
```

Comment:
The string will be returned if it's an instance of the `str` object, meaning, it will not be encoded. I would start by checking that this string is of the type that you are expecting rather than what it's not, if that makes sense?

I also think that an encode function shouldn't decode as that is a separate action.

Reply:
This is from the `elife-bot` project where the combination of these two confusing functions produces consistent output in both Python 2.7 and Python 3. I spent a very long time working on them, and if someone else find a better way, I can make a suggestion of where the test cases are in the `elife-bot` project that must pass when calling these. They were necessary to encode the content back from the dockerised pandoc and compare to the test fixture.

Reply:
I'm not sure if this is relevant but here is some code to handle different versions of python:
```
import sys

if (sys.version_info > (3, 0)):
     # Python 3 code in this block
     import base64
     return base64.b64encode(data).decode()
 else:
     # Python 2 code in this block
     return data.encode("base64")
```